libexec/apply_catalog.rb

```diff
@@ -66,7 +66,9 @@
              Puppet::Transaction::Report.new('apply')
            end
 
-  Puppet.override(current_environment: env, loaders: Puppet::Pops::Loaders.new(env)) do
+  Puppet.override(current_environment: env,
+                  environments: Puppet::Environments::Static.new(env),
+                  loaders: Puppet::Pops::Loaders.new(env)) do
```

Comment:
It's not clear to me that this helps, but it does appear to mimic what 'puppet apply' does.